nvflare/app_opt/job_launcher/k8s_launcher.py

```diff
@@ -0,0 +1,275 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import time
+from enum import Enum
+
+from kubernetes import config
+from kubernetes.client import Configuration
+from kubernetes.client.api import core_v1_api
+from kubernetes.client.rest import ApiException
+
+from nvflare.apis.event_type import EventType
+from nvflare.apis.fl_constant import FLContextKey, JobConstants
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.job_launcher_spec import JobHandleSpec, JobLauncherSpec, JobReturnCode, add_launcher
+from nvflare.apis.workspace import Workspace
+from nvflare.private.fed.utils.fed_utils import extract_job_image
+
+
+class JobState(Enum):
+    STARTING = "starting"
+    RUNNING = "running"
+    TERMINATED = "terminated"
+    SUCCEEDED = "succeeded"
+    UNKNOWN = "unknown"
+
+
+POD_STATE_MAPPING = {
+    "Pending": JobState.STARTING,
+    "Running": JobState.RUNNING,
+    "Succeeded": JobState.SUCCEEDED,
+    "Failed": JobState.TERMINATED,
+    "Unknown": JobState.UNKNOWN,
+}
+
+JOB_RETURN_CODE_MAPPING = {
+    JobState.SUCCEEDED: JobReturnCode.SUCCESS,
+    JobState.STARTING: JobReturnCode.UNKNOWN,
```

Comment:
Should we define JobReturnCode.STARTING? Using UKNOWN is rough.

Reply:
I was thinking this is for job return code, not job status code. that's why define the JobState.STARTING as JobReturnCode.UNKNOWN. BTW, this is only for the K8s launcher which has the JobState.STARTING state. 

But, we may also consider to extend the JobReturnCode with more job states. we can enhance this in the future.